apps/site/lib/site/react/worker.ex

```diff
@@ -27,30 +27,64 @@ defmodule Site.React.Worker do
 
     response =
       ""
-      |> receive_response()
-      |> Poison.decode!()
+      |> receive_response(&handle_json/1)
 
-    if Map.get(response, "error", nil) do
-      {:reply, {:error, response}, state}
+    if get_in(response, [:data]) do
+      if get_in(response, [:data, "error"]) do
+        {:reply, {:error, response.data}, state}
+      else
+        {:reply, {:ok, response.data}, state}
+      end
     else
-      {:reply, {:ok, response}, state}
+      {:noreply, state}
     end
   end
 
-  defp receive_response(prev_responses) do
+  defp receive_response(prev_responses, handle_fn) do
     receive do
       {_, {:data, data}} ->
         str = to_string(data)
-        new_resp = prev_responses <> str
+        resp = prev_responses <> str
+        handle_response(resp, handle_fn)
+    end
+  end
+
+  def handle_response(resp, handle_fn) do
+    {time, result} =
+      :timer.tc(fn ->
+        # Split and keep delimiter
+        ~r/(?<=\n)/
+        |> Regex.split(resp, trim: true)
+        |> Enum.map(&handle_data(&1, handle_fn))
+        |> Enum.find(&(!is_nil(&1)))
+      end)
 
-        if String.ends_with?(str, "\n") do
-          new_resp
-        else
-          receive_response(new_resp)
-        end
+    Logger.warn("node_logging req_time milliseconds=#{time / 1_000}")
```

Comment:
I moved the req time logging here because having it in node was too slow. 

Tests were timing out on {:render, Component, props} intermittently. I think it was pushing certain tests over the threshold for workers (5000ms), some of these renderings are already slow as we know.

We were also getting the huge HTML blob and the final node_logging line in the same stream response so elixir had to do a lot of string parsing.